app/models/name.rb

```diff
@@ -1639,22 +1679,33 @@ def self.parse_author(str)
   end
 
   def self.parse_group(str, deprecated = false)
-    results = nil
-    if match = GROUP_PAT.match(str)
-      name = match[1]
-      text_name = name.tr("ë", "e")
-      parent_name = name.sub(LAST_PART, "")
-      results = ParsedName.new(
-        text_name: text_name + " group",
-        search_name: text_name + " group",
-        sort_name: format_sort_name(text_name, "group"),
-        display_name: format_name(name, deprecated) + " group",
-        parent_name: parent_name,
-        rank: :Group,
-        author: ""
-      )
+    return unless match = GROUP_PAT.match(str)
+
+    # Parse the string without "group"
+    ungrouped_str = str.sub(/\s#{GROUP_ABBR}/, "")
+    result = parse_name(ungrouped_str)
+    return nil unless result
+
+    # Adjust the parsed name
+    result.text_name += " group"
+
+    if result.author.present?
+      # Add "group" before author
+      author = Regexp.escape(result.author)
+      result.search_name.sub!( /(.*)(#{author})/, '\1group \2')
+      result.sort_name.sub!(   /(.*)(#{author})/, '\1 group  \2')
+      result.display_name.sub!(/(.*)(#{author})/, '\1group \2')
+    else
```

Comment:
This approach worries me a bit.  It's hard to predict how a strange author might match some part of the genus or species in very rare cases.  Can we assume that the author has to come at the end?  I think that's safe.  Then we can calculate exactly where to insert "group" in each of search_name, sort_name and display_name.  Or just anchor the regexen to the end of the string, that would also feel safer to me.